### PR TITLE
Removed Content Type header for 304 response since no content is being returned

### PR DESCRIPTION
**Pull request by ChiragRupani**
Summary of the changes
 - Removes Content Type header when preconditional headers process to 304 (NotModified)

Addresses #8230

### CODE REVIEW

File: src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs [hunk 1]
```diff
@@ -75,6 +75,7 @@ internal enum PreconditionState
             if (preconditionState == PreconditionState.NotModified)
             {
                 response.StatusCode = StatusCodes.Status304NotModified;
+                response.Headers.Remove("Content-Type");
```
Comment (pranavkm) [inline src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: @Tratcher does a 412 require the `Content-Type` header to be sent? If not, we could move these two lines https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs#L56-L57 to before here - https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs#L86.

Comment (Tratcher) [reply src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: No, 412 doesn't need it either. Moving SetContentType and SetContentDispositionHeader would be better than removing the value here.

**Comment by pranavkm** [comment]
@ChiragRupani could you make the changes I suggested here: https://github.com/aspnet/AspNetCore/pull/8557#discussion_r266514066? I'd also recommend rebasing on master. We had some build infrastructure issues that are addressed in more recent commits.

File: src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs (hunk 1 above)
Comment (ChiragRupani) [reply src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: Done

File: src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs [hunk 2]
```diff
@@ -74,7 +71,7 @@ internal enum PreconditionState
             // Short circuit if the preconditional headers process to 304 (NotModified) or 412 (PreconditionFailed)
             if (preconditionState == PreconditionState.NotModified)
             {
-                response.StatusCode = StatusCodes.Status304NotModified;
+                response.StatusCode = StatusCodes.Status304NotModified;
```
Comment (pranavkm) [inline src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: Could you undo this change?

File: src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs [hunk 3]
```diff
@@ -83,10 +80,14 @@ internal enum PreconditionState
...
             }
 
+            SetContentType(context, result);
+            SetContentDispositionHeader(context, result);
+
             if (fileLength.HasValue)
             {
                 // Assuming the request is not a range request, and the response body is not empty, the Content-Length header is set to 
                 // the length of the entire file. 
+
```
Comment (pranavkm) [inline src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: Undo please

File: src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs (hunk 2 above)
Comment (ChiragRupani) [reply src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: Removed spaces

File: src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs (hunk 3 above)
Comment (ChiragRupani) [reply src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: Removed spaces

**Review by Tratcher — APPROVED** [review]

**Comment by pranavkm** [comment]
Thanks for the PR!
